src/main/java/com/atlassian/db/replica/api/reason/Reason.java

```diff
@@ -37,12 +41,16 @@ public boolean isRunOnMain() {
         return isRunOnMain;
     }
 
+    public boolean isWrite() {
+        return isWrite;
+    }
+
     @Override
     public boolean equals(Object o) {
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
         Reason reason = (Reason) o;
-        return isRunOnMain == reason.isRunOnMain && Objects.equals(name, reason.name);
+        return isRunOnMain == reason.isRunOnMain && isWrite == reason.isWrite && Objects.equals(name, reason.name);
     }
 
     @Override
```

Comment:
`hashCode` ignores isWrite

Reply:
> return isRunOnMain == reason.isRunOnMain && isWrite == reason.isWrite && Objects.equals(name, reason.name);
I've added isWrite to hashCode today morning,
Do you meant sth else?